### PR TITLE
Audit and Fix `testing-and-debugging.md`:

### PR DESCRIPTION
**Pull request by abatomunkuev**
## What changes are proposed in this pull request?
Solves issue #86 which is a sub-task of issue #18. 

I have audited and fixed an existing Markdown file `docs/testing-and-debugging.md`. 

What have been done: 
- Fixed typos
- Added Frontmatter metadata
- Added and fixed code blocks
- Removed unnecessary backticks, wrapped function names in backticks
- Added alt tags for all images
- Added inter-site links to other pages
- Changed syntax highlighting

## Lighthouse test
image

### CODE REVIEW

File: docs/B-Computations/testing-and-debugging.md [hunk 1]
```diff
@@ -60,7 +64,7 @@ Semantic errors are errors that fail to implement the intent and meaning of the
 Techniques for identifying semantic errors include:
 
 * vocalization - use your sense of hearing to identify the error \(compound conditions\)
-* intermediate output - printf\(\) statements at critical stages
+* intermediate output - `printf()` statements at critical stages
 * walkthrough table
 * interactive debugging using
   * Visual Studio IDE - integrated debugger for Windows OSs
```
Comment (hlavu) [inline docs/B-Computations/testing-and-debugging.md:70]: image
I think you should add a white space before line 71 and 72 for consistency

File: docs/B-Computations/testing-and-debugging.md [hunk 2]
```diff
@@ -286,19 +290,19 @@ When we start gdb, our program pauses.  This is our first opportunity to set bre
 
 The `gdb` commands that we may enter at the prompt include:
 
-* **`list`** ``- lists the 10 lines of source code in the vicinity of where execution has stopped.  Each call advances the current line by 10
+ * **`list`** - lists the 10 lines of source code in the vicinity of where execution has stopped.  Each call advances the current line by 10
 * **`list m, n`** - where m and n are line numbers - lists lines m through n inclusive of the source code.  This call advances the current line to n+1
```
Comment (hlavu) [inline docs/B-Computations/testing-and-debugging.md:294]: Remove redundant white space at line 293 for consistency.

File: docs/B-Computations/testing-and-debugging.md [hunk 3]
```diff
@@ -241,11 +245,11 @@ To trace through execution of our program using the built-in debugger
...
 * Observe the values under the Locals tab in the window below the source code
-* Press F5, note the position of the arrow identifying the next statement to be executed and observe the value of total
+* Press F5, note the position of the arrow identifying the next statement to be executed, and observe the value of total
 * Press F5 and answer the prompt by entering a value of 2
-* Press F5, note the position of the arrow identifying the next statement to be executed and observe the value of total
+* Press F5, note the position of the arrow identifying the next statement to be executed, and observe the value of total
 * Press F5 and answer the prompt by entering a value of 0
-* Press F5, note the position of the arrow identifying the next statement to be executed and observe the value of total
+* Press F5, note the position of the arrow identifying the next statement to be executed, and observe the value of total
 * Press F5 and
```
Comment (hlavu) [inline docs/B-Computations/testing-and-debugging.md:253]: image
1) Add a white space at the beginning of line 256
2) I wonder that is it appropriate to make line 254 and 255 become sub bullet points of line 253 like this:

* Press F5 and
  * Select the command prompt window and observe the program output
  * Select the source code window

File: docs/B-Computations/testing-and-debugging.md (hunk 2 above)
Comment (abatomunkuev) [reply docs/B-Computations/testing-and-debugging.md]: Done

File: docs/B-Computations/testing-and-debugging.md (hunk 1 above)
Comment (abatomunkuev) [reply docs/B-Computations/testing-and-debugging.md:70]: image

On local machine it looks consistent.

**Review by humphd — APPROVED** [review]

**Pull request merged**